backend/src/org/commcare/session/CommCareSession.java

```diff
@@ -192,6 +212,9 @@ public String getNeededData() {
 
         if (needDatum != null) {
             return needDatum;
+        } else if (entries.size() == 1 && entries.elementAt(0) instanceof SyncEntry) {
+            // one sync entry, with no needed data, matches the command; fire that request
```

Comment:
seems like an unnecessary comment that could potentially confuse more than it helps